### PR TITLE
Fluent-theme: clean the fluent styles so no hard coded colors are used

### LINKED ISSUE #7983: Fluent styles should use palette colors
### Environment Information 
 
- __Package version(s)__: Latest
- __Browser and OS versions__: N/A

#### Description
 
The Fluent styles include colors like `CommunicationColors.primary` and `NeutralColors.gray160`, which are **not themeable**. These should be replaced with `palette.themePrimary` and `palette.neutralPrimary`.

We weren't able to do this before because buttons didn't accept a style function, but it should now be possible thanks to #7748.

#### Next steps
- [x] @betrue-final-final to update the designs to remove instances of Gray120, Gray110, and Gray80, which aren't in the neutral palette.
- [x] Update all button components to accept style functions, so we can access the palette.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Normal
 
Products/sites affected: Outlook has run into theming issues

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #7983 
- [X] Include a change request file using `$ npm run change`

#### Description of changes
1. Cleaning all the hard-coded fluent color values with the ones coming from the theme. This required bringing back the `neutralSecondaryAlt` from the deprecation status. All the colors swaps changes were approved by designers (@betrue-final-final ) and @phkuo might have to change the `ThemeGenerator` logic to take into account the `neutralSecondaryAlt` resurrection from death.
2. Make the buttons styles to be functions so that we can access the theme palette and not use the hard-coded colors which were blocking OWA team from testing the package (thanks to #7748). (@MLoughry FYI)
3. Clean up the use of hard-coded `boxShadow` and `borderRadius` values from FluentDepths and `styleConstants` with the ones passed through `theme.effects`
4. Adjust the types on `IEffects` properties to allow more flexibility for what to pass and making it easier to use in styles. (discussed with @phkuo )

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8098)

### CODE REVIEW

File: packages/fluent-theme/src/fluent/FluentTheme.ts [hunk 1]
```diff
@@ -8,14 +9,25 @@ export const FluentTheme: ITheme = createTheme({
...
     neutralTertiaryAlt: NeutralColors.gray60,
     neutralQuaternary: NeutralColors.gray50,
     neutralQuaternaryAlt: NeutralColors.gray40,
     neutralLight: NeutralColors.gray30,
     neutralLighter: NeutralColors.gray20,
     neutralLighterAlt: NeutralColors.gray10,
-    white: NeutralColors.white
+    white: NeutralColors.white,
+    // Shared Colors
+    red: SharedColors.red10,
```
Comment (phkuo) [inline packages/fluent-theme/src/fluent/FluentTheme.ts]: these only need to be added if they're different than the default

File: packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts [hunk 2]
```diff
@@ -1,13 +1,18 @@
 import { IBasePickerStyleProps, IBasePickerStyles } from 'office-ui-fabric-react/lib/Pickers';
-import { fluentBorderRadius } from './styleConstants';
 
 export const BasePickerStyles = (props: IBasePickerStyleProps): Partial<IBasePickerStyles> => {
+  const { theme } = props;
+  if (!theme) {
+    throw new Error('Theme is undefined or null.');
```
Comment (phkuo) [inline packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts]: shouldn't this never happen? isn't a base part of fabric that we guarantee the theme?

File: packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts [hunk 3]
```diff
@@ -33,7 +31,7 @@ export const CheckboxStyles = (props: ICheckboxStyleProps): Partial<ICheckboxSty
         !checked && {
           selectors: {
             ':hover .ms-Checkbox-text': { color: palette.neutralDark },
-            ':hover .ms-Checkbox-checkmark': { color: NeutralColors.gray120 } // color not in the palette or semanticColors
+            ':hover .ms-Checkbox-checkmark': { color: palette.neutralSecondary }
```
Comment (phkuo) [inline packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts]: i don't know if you want to address this or do it in another PR, but if this style is already using neutralSecondary in the base component, then it is not needed here

Comment (phkuo) [reply packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts]: (and same with all over the place. it would be preferable to remove as many styles as possible to keep the bundle size regression as small as possible...)

File: packages/styling/src/interfaces/IEffects.ts [hunk 4]
```diff
@@ -10,28 +10,28 @@ export interface IEffects {
    * Usually a shadow. This is lower than elevations with a higher value, and higher than elevations with a lower value.
    * Used for: cards, grid items
    */
-  elevation4: IRawStyle;
+  elevation4: string;
   /**
    * Used to provide a visual affordance that this element is elevated above the surface it rests on.
    * Usually a shadow. This is lower than elevations with a higher value, and higher than elevations with a lower value.
```
Comment (phkuo) [inline packages/styling/src/interfaces/IEffects.ts]: remove the "usually a shadow" string from comments here

File: packages/fluent-theme/src/fluent/FluentTheme.ts (hunk 1 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/FluentTheme.ts]: By these, you mean the red colors or the white too? cause technically the white and black are the same 😄

Comment (phkuo) [reply packages/fluent-theme/src/fluent/FluentTheme.ts]: white and black too! might as well keep it slim

File: packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts (hunk 2 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts]: Some of the components don't follow this pattern in their style props to make the `theme` required and this happens. I believe that is something that needs to be cleaned up as well to have a consistent API across all components.

File: packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts (hunk 3 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts]: Good point on that, but this PR is to unblock the testing of the package by other teams so I might take another look at compared to the default styles with a separate PR. Or maybe over the weekend in this PR. Thanks.

File: packages/fluent-theme/src/fluent/FluentTheme.ts (hunk 1 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/FluentTheme.ts]: Will clean the white and black but the reds are differing from the default colors so I need these ones in here.

File: packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts (hunk 3 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/styles/Checkbox.styles.ts]: Cleaned as much as possible without breaking anything. Don't think it's worth the time going over again as will move them to default styles and will make sure there are no redundant styles plus don't think I can trim a sensible amount off to make a dent in bundle size.

File: packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts (hunk 2 above)
Comment (JasonGore) [reply packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts]: Whether or not theming is required via props typing doesn't affect whether it's actually available. This is actually determined by the use of `styled` helper which will automatically inject `themes` prop, even if the component has made `theme` prop optional. I've modified this approach in `Foundation` by making theme prop required as part of an inherited interface.

For now, no other components are actually checking for theme's existence in styles functions, so I think it's consistent to assume it's available in styles functions.

Comment (JasonGore) [reply packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts]: (The assumption here is that 'some of the components' are all components using `styled`. Components that use `customizable` or other methods cannot technically have styles functions called.)

File: packages/fluent-theme/src/fluent/styles/Callout.styles.ts [hunk 5]
```diff
@@ -1,19 +1,20 @@
...
+  const { theme } = props;
+  const { effects } = theme;
+
   return {
     root: {
-      borderRadius: fluentBorderRadius,
+      borderRadius: effects.roundedCorner2,
       borderWidth: 0,
-      boxShadow: Depths.depth16
+      boxShadow: effects.elevation16
```
Comment (mikewheaton) [inline packages/fluent-theme/src/fluent/styles/Callout.styles.ts]: So great to see this is in the theme now!

**Comment by Vitalius1** [comment]
@Jahnp I did look through the components both during making the changes and after I've done them. There is still one more PR I want to submit with passing the font type ramp through the theme and after will need to make another full pass through the components in pair with Ben to make sure we have followed the redlines in full (minus the components that are still waiting for design).

File: packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts (hunk 2 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts]: @JasonGore if I remove the theme check on this components I get this lint message:
screen shot 2019-02-25 at 10 50 11 am
Also, there are a couple of components in OUFR that are still doing the same thing. And I know that in Foundation this is fixed 👍  Thanks for the consisitency in the API that it will bring to our repo.

**Comment by Vitalius1** [comment]
@dzearing do you mind to take a look, please?

File: packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts (hunk 2 above)
Comment (JasonGore) [reply packages/fluent-theme/src/fluent/styles/BasePicker.styles.ts]: Yes, this is because of the typing on the component making it optional. Other styles functions do this:
```tsx
const { effects } = theme!;
```

Which of course isn't ideal either. In either case an exception would occur. Maybe your approach is best for communicating the actual issue.
